designsystem/src/desktopTest/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputPhoneNumberTest.kt

```diff
@@ -0,0 +1,124 @@
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").performTextInput("1111")
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").assertTextEquals("1111")
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").performTextInput("1111a")
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").assertTextEquals("1111")
+    }
+
+    @Test
+    fun shouldEnableCallActionButtonAfterInputTextReachesCharacterLimit() {
+        rule.setContent {
+            var inputValue by remember { mutableStateOf("") }
+
+            InputPhoneNumber(
+                title = "Phone Number",
+                inputText = inputValue,
+                onValueChanged = {
+                    if (it != null) {
+                        inputValue = it
+                    }
+                },
+                onCallActionClicked = {
+                    // no-op
+                },
+            )
+        }
+        rule.onNodeWithTag("CALL_PHONE_NUMBER_BUTTON").assertIsNotEnabled()
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").performTextInput("1111111111")
+        rule.onNodeWithTag("CALL_PHONE_NUMBER_BUTTON").assertIsEnabled()
+    }
+
+    @Test
+    fun shouldDisplaySupportTextIfInputStateIsError() {
+        rule.setContent {
+            InputPhoneNumber(
+                title = "Phone Number",
+                inputText = "",
+                state = InputShellState.UNFOCUSED,
+                onValueChanged = {
+                    // no-op
+                },
+                onCallActionClicked = {
+                    // no-op
+                },
+            )
+        }
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_SUPPORTING_TEXT").assertDoesNotExist()
+
+        rule.setContent {
+            InputPhoneNumber(
+                title = "Phone Number",
+                inputText = "",
+                state = InputShellState.ERROR,
+                onValueChanged = {
+                    // no-op
+                },
+                onCallActionClicked = {
+                    // no-op
+                },
+            )
+        }
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_SUPPORTING_TEXT").assertExists()
+    }
+
+    @Test
+    fun shouldClearPhoneNumberWhenClearButtonIsClicked() {
+        rule.setContent {
+            var inputValue by remember { mutableStateOf("1234") }
+
+            InputPhoneNumber(
+                title = "Phone Number",
+                inputText = inputValue,
+                onValueChanged = {
+                    if (it != null) inputValue = it
+                },
+                onCallActionClicked = {
+                    // no-op
+                },
+            )
+        }
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_RESET_BUTTON").assertExists()
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_RESET_BUTTON").performClick()
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_FIELD").assertTextEquals("")
+        rule.onNodeWithTag("INPUT_PHONE_NUMBER_RESET_BUTTON").assertDoesNotExist()
+    }
```

Comment:
Could you make a test also for a prefix, like _+34 654342555_?

Reply:
At the moment the component doesn't do the prefix formatting or anything like that. Existing regex setup only allows us to validate and invoke callback if entire string matches.

Our other option is to skip validation until the character limit is reached for phone number and then do formatting as well.

For now the regex is purely to set which characters to allow, at the moment it allows digits and `+` sign.

Reply:
With this, we don't particularly need a new test as we are evaluating content is actually changing. We are not doing formatting yet, so we can skip it for now. WDYT?

Reply:
yes, I agree we should not validate and neither formatting right now, the regex is just to allowed characters, not for validation, great!. I was just suggesting to do one with prefix in order to be sure that we allow the "+" character. But it is OK